src/presign/participant.rs

```diff
@@ -686,20 +745,41 @@ impl PresignParticipant {
     ) -> Result<Vec<Message>> {
         info!("Generating round three presign messages.");
 
-        let keyshare = get_keyshare(self.id, input)?;
-        let round_three_hashmap = self.get_other_participants_round_three_values(input)?;
+        let info = PresignKeyShareAndInfo::new(self.id, input)?;
+        // Collect the other participant's values from storage needed for round
+        // three.
+        //
+        // XXX remove these hashmaps, both here and elsewhere. It'd be much more
+        // straightforward to just pass around vectors.
```

Comment:
I agree. I recently added a participant ID to `AuxinfoPublic` so we don't even need to save that pid here.

I would actually guess that if we refactor the round three protocol steps back into this module (#353) we won't need to make any new data structures, and can just get the fields from storage as we need them.

Reply:
I added notes to #351 and #353 to address this issue as part of that refactor. That should cover this concern.